src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -3389,6 +3389,12 @@ public ILocalSymbol GetDeclaredSymbol(CatchDeclarationSyntax catchDeclaration)
                         {
                             if (resultKind == LookupResultKind.NotCreatable)
                             {
+                                if (expr.Type is TypeParameterSymbol)
+                                {
+                                    resultKind = LookupResultKind.Empty;
+                                    return symbols;
+                                }
```

Comment:
Feels like we should separate this out into a different PR. It's not related to the `??=` change.

Reply:
@DoctorKrolic did you open a separate PR for this? Would prefer to merge that first.

Reply:
Please take a look: https://github.com/dotnet/roslyn/pull/74675